src/magic_folder/config.py

```diff
@@ -214,16 +216,17 @@
         )
         """,
         """
-        -- This table represents the current of the file on disk, as last known to us
+        -- This table represents the current state of the file on disk, as last known to us
         CREATE TABLE [current_snapshots]
         (
-            [name]          TEXT PRIMARY KEY, -- mangled name in UTF-8
-            [snapshot_cap]  TEXT,             -- Tahoe-LAFS URI that represents the most recent remote snapshot
-                                              -- associated with this file, either as downloaded from a peer
-                                              -- or uploaded from local changes
-            [mtime_ns]      INTEGER NOT NULL, -- ctime of current snapshot
-            [ctime_ns]      INTEGER NOT NULL, -- mtime of current snapshot
-            [size]          INTEGER NOT NULL  -- size of current snapshot
+            [name]             TEXT PRIMARY KEY, -- mangled name in UTF-8
+            [snapshot_cap]     TEXT,             -- Tahoe-LAFS URI that represents the most recent remote snapshot
+                                                 -- associated with this file, either as downloaded from a peer
+                                                 -- or uploaded from local changes
+            [mtime_ns]         INTEGER NOT NULL, -- ctime of current snapshot
+            [ctime_ns]         INTEGER NOT NULL, -- mtime of current snapshot
+            [size]             INTEGER NOT NULL, -- size of current snapshot
+            [last_updated_ns]  INTEGER NOT NULL  -- timestamp when last changed
```

Comment:
I'm slightly tempted to say this should not be in ns, given that we don't care about that level of accuracy for it, but consistency is probably more valuable.

Reply:
I don't think it's a good idea for _any_ of these to be nanoseconds (for the same reason) but yes, consistency is better.

Reply:
There is good reason for mtime/ctime to be nanoseconds, although when I added those values, I never meant them to be exposed to users.  I wonder if it would make sense to track the user-visible modified time separately in seconds (along with last-updated time)? The `PathState` tuple was really intended to be a mostly opaque snapshot of (part of) the file's stat-info, that we can use to notice when the file has changed; for that use case, we want as much accuracy as possible.

Reply:
Users do not see nanoseconds.

Reply:
I ... know. My suggestion was to decouple the user-visible data we track, from the internal state tracking. It isn't obvious that they always want to be in sync. For example, if somebody `touch`s a file[1], we must update the internal state tracking, but we might want to not update the user-visible modified time, given that the file wasn't actually modified. Just because the two values happen to be equal now, doesn't mean we should store them in the same column, if they have different meanings.

[1] To be extra clear, we *don't* do this now, and we may not want to do this in the future. But I think it is entirely reasonable to think about potential future cases, when thinking about how to model data.

Reply:
And, to be clear, neither of the points are blocking landing this.